dashboard/app/helpers/users_helper.rb

```diff
@@ -1,2 +1,59 @@
 module UsersHelper
+
+  # Summarize the current user's progress within a certain script.
+  def summarize_user_progress(script, callouts_seen = nil, videos_seen = nil)
+    if (!current_user)
+      lines = session[:lines] || 0
+      user_data = {
+        linesOfCode: lines,
+        linesOfCodeText: t('nav.popup.lines', lines: lines),
+      }
+
+      # Check all levels for progress
+      levels = script.script_levels
+    else
+      user_data = {
+        linesOfCode: current_user.total_lines,
+        linesOfCodeText: t('nav.popup.lines', lines: current_user.total_lines),
+        levels: {},
+        videos_seen: videos_seen,
+        callouts_seen: callouts_seen
+      }
+
+      user_data[:disableSocialShare] = true if current_user.under_13?
+
+      if script.trophies
+        progress = current_user.progress(script)
```

Comment:
since we throw away current_levels and max_levels from User#progress maybe we should call another method that doesn't do that query.

(does anything even use current_levels and max_levels? A quick grep doesn't show anything.)